### PR TITLE
Max rather than sum for included suggestion relevance boost

### PR DESCRIPTION
**Pull request by lambdabaa**
From the [API spec](https://htmlpreview.github.io/?https://github.com/dart-lang/sdk/blob/master/pkg/analysis_server/doc/api.html):

> If an AvailableSuggestion has relevance tags that match more than one
> IncludedSuggestionRelevanceTag, the maximum relevance boost is used.

/cc @alexander-doroshko @scheglov @DanTup

### CODE REVIEW

File: Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java [hunk 1]
```diff
@@ -549,7 +549,7 @@ private static CompletionSuggestion createCompletionSuggestionFromAvailableSugge
       for (String tag : relevanceTags) {
         IncludedSuggestionRelevanceTag relevanceTag = includedSuggestionRelevanceTags.get(tag);
         if (relevanceTag != null) {
-          relevance += includedSuggestionRelevanceTags.get(tag).getRelevanceBoost();
+          relevance = Math.max(relevance, relevanceTag.getRelevanceBoost());
```
Comment (scheglov) [inline Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java]: Hm... "the maximum relevance *boost* is used".
Which means that the boost is *added* to the base relevance.

Comment (DanTup) [reply Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java]: This was my understanding too. It should be `base relevance + (maximum of tag boosts)`.

Comment (lambdabaa) [reply Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java]: Cool makes sense, updated!

Comment (alexander-doroshko) [reply Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java]: @scheglov, is it safe to start with zero assuming that there always be at least one non-negative relevance boost?

@lambdabaa, there are some not related files in the commit :)

Comment (lambdabaa) [reply Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java]: Thanks for pointing that out. `git commit --amend` not actually the same as `git rebase -i HEAD~2`...

Comment (lambdabaa) [reply Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java]: > @scheglov, is it safe to start with zero assuming that there always be at least one non-negative relevance boost?

There may be no relevance boosts but then 0 boost is what we want.

**Comment by lambdabaa** [comment]
Moved to #660.

File: Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java (hunk 1 above)
Comment (scheglov) [reply Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java]: Yes, there are no negative boosts.

**Pull request closed without merging**